File: app/src/main/java/com/ooftf/demo/layout_chain/demo4/MyAdapter.java
package com.ooftf.demo.layout_chain.demo4;

import android.graphics.Color;
import android.util.SparseIntArray;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.Random;

public class MyAdapter extends StickyAdapter<MyAdapter.MyViewHolder> {
    Random random = new Random();
    SparseIntArray sia = new SparseIntArray();
    SparseIntArray h = new SparseIntArray();
    {
        for (int i = 0; i < 1000; i++) {
            if(i == getStickyPosition()){
                sia.put(i,0);
                h.put(i,200);
            }else{
                sia.put(i,random.nextInt(2));
                h.put(i,(random.nextInt(2)+1)*100);
            }


        }
    }

    public MyAdapter(@NonNull StickyRecyclerViewLayout layout) {
        super(layout);
    }


    /**
     * 返回 item 个数
     * @return
     */
    @Override
    public int getItemCount() {
        return 1000;
    }


    @Override
    public void onBindViewHolderEx(@NonNull MyViewHolder holder, int position) {
        ((TextView)(holder.itemView)).setHeight(h.get(position));
        holder.itemView.setBackgroundColor(Color.argb(255,random.nextInt(255),random.nextInt(255),random.nextInt(255)));
        ((TextView)(holder.itemView)).setText("i::"+position);
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolderEx(@NonNull ViewGroup parent, int viewType) {
        TextView textView = new TextView(parent.getContext());
        textView.setPadding(10,10,10,10);
        textView.setGravity(Gravity.CENTER);
        return new MyViewHolder(textView);
    }

    /**
     * 返回 粘性布局的位置
     * @return
     */
    @Override
    public int getStickyPosition() {
        return 30;
    }

    /**
     * 返回view 的类型，所有 card 类型都是一样的，其他 根据 数据中 layout.type 类型的不同返回不同的值
     * @param position
     * @return
     */
    @Override
    public int getItemViewType(int position) {
        return super.getItemViewType(position);
    }

    @Override
    public boolean isFullSpan(int position) {
        return sia.get(position) == 0;
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder{

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/ooftf/demo/layout_chain/demo4/MyAdapter.kt
package com.ooftf.demo.layout_chain.demo4

import android.graphics.Color
import com.ooftf.basic.utils.DensityUtil.dip2pxInt
import android.view.ViewGroup
import androidx.core.view.NestedScrollingParent
import com.ooftf.basic.utils.DensityUtil
import com.ooftf.demo.layout_chain.demo2.CoordinatorLayout.ProgressChangeListener
import android.view.ViewGroup.MarginLayoutParams
import androidx.core.view.ViewCompat
import android.widget.OverScroller
import androidx.core.view.ScrollingView
import android.widget.FrameLayout
import androidx.recyclerview.widget.RecyclerView
import android.util.SparseIntArray
import android.widget.TextView
import android.view.Gravity
import com.ooftf.demo.layout_chain.demo3.PageView
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import android.view.MotionEvent
import android.view.View
import com.ooftf.demo.layout_chain.demo4.StickyRecyclerViewLayout
import com.ooftf.demo.layout_chain.demo4.StickyAdapter
import java.util.*

class MyAdapter(layout: StickyRecyclerViewLayout) : StickyAdapter<MyAdapter.MyViewHolder>(layout) {
    var random = Random()
    var sia = SparseIntArray()
    var h = SparseIntArray()

    /**
     * 返回 item 个数
     * @return
     */
    override fun getItemCount(): Int {
        return 1000
    }

    override fun onBindViewHolderEx(holder: MyViewHolder, position: Int) {
        (holder.itemView as TextView).height = h[position]
        holder.itemView.setBackgroundColor(
            Color.argb(
                255,
                random.nextInt(255),
                random.nextInt(255),
                random.nextInt(255)
            )
        )
        holder.itemView.text = "i::$position"
    }

    override fun onCreateViewHolderEx(parent: ViewGroup, viewType: Int): MyViewHolder {
        val textView = TextView(parent.context)
        textView.setPadding(10, 10, 10, 10)
        textView.gravity = Gravity.CENTER
        return MyViewHolder(textView)
    }

    /**
     * 返回 粘性布局的位置
     * @return
     */
    override fun getStickyPosition(): Int {
        return 30
    }

    /**
     * 返回view 的类型，所有 card 类型都是一样的，其他 根据 数据中 layout.type 类型的不同返回不同的值
     * @param position
     * @return
     */
    override fun getItemViewType(position: Int): Int {
        return super.getItemViewType(position)
    }

    override fun isFullSpan(position: Int): Boolean {
        return sia[position] == 0
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    init {
        for (i in 0..999) {
            if (i == getStickyPosition()) {
                sia.put(i, 0)
                h.put(i, 200)
            } else {
                sia.put(i, random.nextInt(2))
                h.put(i, (random.nextInt(2) + 1) * 100)
            }
        }
    }
}